### PR TITLE
[Feature] First draft of populated sets

### PR DESCRIPTION
**Pull request by rvyuha**
The changed file contains the first draft of sets

### CODE REVIEW

**Review by yulric — CHANGES_REQUESTED** [review]
1. Can we create a function that wraps li and ul tags within a string. It looks like we'll be using it in multiple reference list items.

File: qmd/sets.qmd [hunk 1]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
...
+handle_empty_character_vector <-
+  function(check_vector, replace_string) {
+    if (length(check_vector) < 1) {
+      return(replace_string)
+    }else{
+      return(check_vector)
+    }
+  }
+
+list_creation <- function(string_to_append, list_values){
```
Comment (yulric) [inline qmd/sets.qmd:17]: Rename to `create_html_list`. Functions should start with verbs.

Comment (yulric) [reply qmd/sets.qmd:17]: I don't think the appending needs to be done in this function i.e. we don't need the `string_to_append` parameter,

A function should try to have only one responsibility and there's no need for this function to have the responsibility of the appending.

Comment (yulric) [reply qmd/sets.qmd:17]: Add Roxygen comment.

File: qmd/sets.qmd [hunk 2]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
...
+  string_to_append <- glue::glue('{string_to_append}
```
Comment (yulric) [inline qmd/sets.qmd:18]: We can insert both ul tags at the end just like the li tags in line 21-22

File: qmd/sets.qmd [hunk 3]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
 
 
 ```{r, echo=FALSE}
+# Utility Functions
+handle_empty_character_vector <-
```
Comment (yulric) [inline qmd/sets.qmd:8]: Rename to `replace_if_empty_character_vector`

File: qmd/sets.qmd [hunk 4]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
...
+  function(check_vector, replace_string) {
```
Comment (yulric) [inline qmd/sets.qmd:9]: Rename to `vector_to_check`. Variable names should not start with verbs.

Comment (yulric) [reply qmd/sets.qmd:9]: Rename to `str_replace`. Variable names should not start with verbs.

File: qmd/sets.qmd [hunk 5]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
...
-                                <ul>')
+                                ### {set_ID} set members:')
+    list_values <- list()
     
     # Find and loop over unique partID according to set_ID
-    sub_category_rows <- working_rows[working_rows$setID == set_ID,]
-    unique_part_id <- unique(sub_category_rows$partID)
-    for (part_ID in unique_part_id) {
+    current_set_ID_rows <- rows_for_current_set_type[rows_for_current_set_type$setID == set_ID,]
+    unique_part_IDs <- unique(current_set_ID_rows$partID)
```
Comment (yulric) [inline qmd/sets.qmd:75]: Rename to `unique_set_ID_rows`

File: qmd/sets.qmd [hunk 6]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
...
+    for (part_ID in unique_part_IDs) {
       part_info <- parts_table[parts_table$partID == part_ID,]
```
Comment (yulric) [inline qmd/sets.qmd:77]: Rename to `set_info`

File: qmd/sets.qmd [hunk 7]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
...
       part_label <- part_info$partLabel
+      part_label <- handle_empty_character_vector(part_label, "Part Label is missing at the moment")
```
Comment (yulric) [inline qmd/sets.qmd:79]: Rename to `set_label`

File: qmd/sets.qmd [hunk 8]
```diff
@@ -4,67 +4,84 @@ The list of sets in aphabetical order of setsID
...
+  rows_for_current_set_type <- sets_table[sets_table$setType == set_type,]
+  unique_set_IDs <- unique(rows_for_current_set_type$setID)
+  for (set_ID in unique_set_IDs) {
     # Create set_ID subsection
     output_string <- glue::glue('{output_string}
     
-                                ### {set_ID} set members:
-                                <ul>')
+                                ### {set_ID} set members:')
+    list_values <- list()
```
Comment (yulric) [inline qmd/sets.qmd:71]: Rename to `curent_set_type_content`

**Review by yulric — CHANGES_REQUESTED** [review]

**Comment by DougManuel** [comment]
`setType` — see example below.

- Change to Header #2. 
- partLabel only for the header.
- Add anchor (the anchor should be added automatically).
- partID in the same format as for Tables; on a separate line.

Screenshot 2023-02-14 at 4 42 18 PM

**Comment by DougManuel** [comment]
For `setID`. See example below.

- header 3.
- with anchor.
- setID label only. for `booleanAggrSet` the label is `Boolean aggregation set`.
- remove `set members:`
- Format similar to headers in tables.
     - bullets.
     - setLabel (partLabel) in bold. 
     - followed by part ID in parenthesis `()` and a period `.`
     - followed by the setID partDesc
     - followed by: 'Status:' (followed by the status 'active' or 'depreciated'. 'First released:' (followed by release version). 'Last updated:' followed by last updated.

Put together the example below would be:

--------


## Aggregation set
partID: [aggSet](https://docs.phes-odm.org/parts.html#aggset). Sets of aggregations that can be used for measures. Examples of aggregation sets include logarithm, linear and boolean.

### Boolean aggregation set
- **Single** (sin). This value is not an aggregate measurement in any way (ie. not a mean, median, max or any other) and can be a replicate value. Status: active. First released: 2.0.0. Last updated: 2.0.0.


--------







Screenshot 2023-02-14 at 4 47 38 PM

**Review by DougManuel — CHANGES_REQUESTED** [review]
@rvyuha I made changes to the formatting in the comments section. 

Could you modify with those changes and then I'll review further?

**Comment by rvyuha** [comment]
@yulric and @DougManuel ready for further review

**Comment by rvyuha** [comment]
@yulric Ready for review

**Comment by DougManuel** [comment]
1. The organization with the listSet is good. But please put in alphabetical order within the listSet header. 

1. However, you are missing those lists within the listSet. For example, you are missing the first 'dataType' list. We want lists for everything in the sets tab (setTypes except dictSet) AND all lists for list generated when you select for partType for the partIDs in listSet.

1. In the example below. Could you:
      - Put a space between ​compartment​ and ​(air)​ 
      - Make a link for ​air​ to its partID in the parts.html page. 
      - Put `Status: active. First released....`on a new line.
  
Screenshot 2023-02-21 at 3 48 37 PM

**Pull request merged**